### PR TITLE
Added list subheader and divider component and updated list styles

### PR DESCRIPTION
**Pull request by jtrujill**
This is a precursor to implementing #63 and #61

### CODE REVIEW

File: lib/src/divider/divider.component.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+$hc-divider-width: 1px;
+
+.hc-divider {
+    display: block;
+    margin: 0;
```
Comment (andrew-frueh) [inline lib/src/divider/divider.component.scss]: Recommend changing this to `margin: 10px 0;` so we have a little more whitespace between sections

File: lib/src/list/list.component.scss [hunk 2]
```diff
@@ -1,3 +1,96 @@
...
+}
+
+.hc-list-subheader {
+    display: flex;
+    box-sizing: border-box;
+    align-items: center;
+    padding: $hc-list-side-padding;
+
+    .hc-list & {
+        margin: 0;
```
Comment (andrew-frueh) [inline lib/src/list/list.component.scss]: Recommend adding `padding-bottom: 3px;` here to tighten up the space between the header and the first list item

File: lib/src/divider/divider.component.scss (hunk 1 above)
Comment (jtrujill) [reply lib/src/divider/divider.component.scss]: I'm using the padding between lists to have whitespace between elements. By setting the margin to 0, I ensure that any element that has a margin defined will be set to 0. This makes the elements more uniform

File: lib/src/list/list.component.scss (hunk 2 above)
Comment (jtrujill) [reply lib/src/list/list.component.scss]: Same as above padding is being used for "whitespace" between elements. Margin: 0 ensures that elements that have the directive hcListSubheader defined don't interfere with the "whitespace" of list elements

**Comment by jtrujill** [comment]
@andrew-frueh I rebased the latest changes from master so that this will now merge without issue

**Pull request closed without merging**